// Copyright (C) 2017 The Android Open Source Project
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";  // use required fields, which aren't in proto3.

package iorap.serialize.proto;  // C++ namespace iorap::serialize::proto
option java_package = "com.google.android.iorap";
option optimize_for = LITE_RUNTIME;

// TODO: should these fields be 'packed' for "smaller encoding" ?

message TraceFile {
  required TraceFileIndex index = 1;
  required TraceFileList list = 2;
}

message TraceFileIndex {
  repeated TraceFileIndexEntry entries = 1;
}

message TraceFileIndexEntry {
  required int64 id = 1;
  required string file_name = 2;
}

message TraceFileList {
  repeated TraceFileEntry entries = 1;
}

message TraceFileEntry {
  required int64 index_id = 1;
  required int64 file_offset = 2;
  required int64 file_length = 3;
}

// XX: use nested messages?